src/test/ui/issue-27282-move-ref-mut-into-guard.stderr

```diff
@@ -0,0 +1,9 @@
+error[E0507]: cannot move out of borrowed content
+  --> $DIR/issue-27282-move-ref-mut-into-guard.rs:24:18
+   |
+LL |             if { (|| { let bar = foo; bar.take() })(); false } => {},
+   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
```

Comment:
I haven't dug too much on the desugaring code, but it seems to me that there could be a way to use `foo`'s span for this error, which would be more appropriate for this and the following error.